packages/subgraph/src/kpi-tokens-manager.ts

```diff
@@ -183,3 +222,92 @@ export function handleUpgradeTemplate(event: UpgradeTemplateEvent): void {
     }
     newTemplate.save();
 }
+
+export function handleSetFeatureSetOwner(event: SetFeatureSetOwnerEvent): void {
+    const oraclesManager = getKPITokensManager();
+    const templateSet = getTemplateSet(
+        bytesToAddress(oraclesManager.id),
+        event.params.templateId,
+    );
+    templateSet.featuresOwner = event.params.owner;
+    templateSet.save();
+}
+
+export function handleEnableFeatureFor(event: EnableFeatureForEvent): void {
+    const feature = getTemplateFeature(
+        event.params.templateId,
+        event.params.featureId,
+    );
+
+    const allowedFeatureAccountId = feature.id.concat(
+        addressToBytes(event.params.account),
+    );
+    let allowed = KPITokenTemplateFeatureAllowedAccount.load(
+        allowedFeatureAccountId,
+    );
+    if (allowed !== null) {
+        log.warning(
+            "tried to double enable feature with id {} on template with id {} for user {}",
+            [
+                event.params.featureId.toString(),
+                templateId.toString(),
+                event.params.account.toString(),
+            ],
+        );
+        return;
+    }
+
+    allowed = new KPITokenTemplateFeatureAllowedAccount(
+        allowedFeatureAccountId,
+    );
+    allowed.feature = feature.id;
+    allowed.address = addressToBytes(event.params.account);
+    allowed.save();
+}
+
+export function handleDisableFeatureFor(event: DisableFeatureForEvent): void {
+    const feature = getTemplateFeature(
+        event.params.templateId,
+        event.params.featureId,
+    );
+    const allowedFeatureAccountId = feature.id.concat(
+        addressToBytes(event.params.account),
+    );
+    if (
+        KPITokenTemplateFeatureAllowedAccount.load(allowedFeatureAccountId) ===
+        null
+    ) {
+        log.warning(
+            "tried to disable feature with id {} on template with id {} for user {} that did not have the feature enabled in the first place",
+            [
+                event.params.featureId.toString(),
+                templateId.toString(),
+                event.params.account.toString(),
+            ],
+        );
+        return;
+    }
+
+    store.remove(
+        "KPITokenTemplateFeatureAllowedAccount",
+        allowedFeatureAccountId.toString(),
+    );
+}
+
+export function handlePauseFeature(event: PauseFeatureEvent): void {
+    const feature = getTemplateFeature(
```

Comment:
Should we handle the case where the feature being paused doesn't exist?

Reply:
In that case, as it is now, it is created and then paused. The contract has no check on this and it happens there too. To be honest I don't think there's any security risk to this, the worst that can happen is that by mistake a certain feature is paused while it shouldn't. At the end of the day a feature is just a mapping from an id to a bool flag.